helpers.go

```diff
@@ -71,12 +71,23 @@ func DecodeResponse(lines []string, v interface{}) error {
 	for _, part := range strings.Split(lines[0], "|") {
 		for _, val := range strings.Split(part, " ") {
 			parts := strings.SplitN(val, "=", 2)
-			// TODO(steve): support groups
 			key := Decode(parts[0])
 			if len(parts) == 2 {
 				v := Decode(parts[1])
 				if i, err := strconv.Atoi(v); err != nil {
-					input[key] = v
+					// Only support comma seperated lists
+					// by keyname to avoid incorrect decoding.
+					if key == "client_servergroups" {
+						var serverGroups []int
+						for _, s := range strings.Split(v, ",") {
+							if i, err := strconv.Atoi(s); err == nil {
+								serverGroups = append(serverGroups, i)
+							}
+						}
+						input[key] = serverGroups
```

Comment:
If these are expected to be int's I would return an error if decode fails.

You can also optimise by using the split size.
```suggestion
						var serverGroups []int
						parts := strings.Split(v, ",")
						serverGroups := make([]int, len(parts))
						for i, s := range parts  {
							group, err := strconv.Atoi(s)
							if err != nil {
								return fmt.Errorf("decode server group: %w", err)						
							}
							serverGroups[i] = group
						}
						input[key] = serverGroups
```

Reply:
Thank you for your insights, I have indeed forgotten to return an error. I believe it is correct that these would indeed be expected as int.
I don't really know if I understand the optimalisation correctly: the benefit here is that the array `serverGroups` is an array made just large enough instead of a dynamic length `[]int` array? Or is the benefit coming from the fact that the split is done outside of the range rule? Or are there some other factors at play here? I don't seem to be sure.
I have adopted your code in the new commit ,except for the first line, and it works for me.

Reply:
Correct as `strings.Split` returns the exact size slice we need when we make `serverGroups` to that size, so the runtime can avoid either creating a slice that's too big wasting memory or creating one too small and having the performance penalty of having  to resize and possibly relocate the data.

In addition to that as we know we have the right size, we can assign by index avoiding the overhead of `append`

Reply:
Thank you for explaining this to me.